kuma/users/views.py

```diff
@@ -477,11 +477,16 @@ def get_form(self, form_class=None):
         self.email_addresses = {}
         form = super(SignupView, self).get_form(form_class)
         form.fields['email'].label = _('Email address')
+
+        # When no username is provided, default to the local-part of the email address
+        if form.initial.get('username', '') == '':
+            form.initial['username'] = form.initial.get('email', ('',))[0].split('@')[0]
```

Comment:
Ah! This is why my default username input became "p". I speculate that `form.initial['emai'] == 'pbengtsso@mozilla.com'` but this code here expects it to be `form.initial['emai'] == ['pbengtsso@mozilla.com']`.

Reply:
Yeah, I put a print on `form.initial` here, deleted myself, and started over. It yields:
```
{'email': 'pbengtsson@mozilla.com', 'username': '', 'first_name': 'Peter', 'last_name': 'Bengtsson'}
```
therefore `form.initial.get('email', ('',))[0].split('@')[0]` will yield `'p'`.

Reply:
Weird, I can see that behavior now too. I actually changed that to a tuple after `.get(` started returning a tuple after the Python 3 upgrade, which I thought was an API change. But now I'm getting the same behavior as you, so I reverted it.